specs/CustomDownload.md

```diff
@@ -0,0 +1,783 @@
+/// `ICoreWebView2Download.State` property to get the current state, which can
+/// be in progress, interrupted, or complete. Find the `InterruptReason` property
+/// on `ICoreWebView2DownloadStateChangedEventArgs`.
+[uuid(74EEBAA9-704B-4A9A-8331-AEBD541CB62C), object, pointer_default(unique)]
+interface ICoreWebView2DownloadStateChangedEventHandler : IUnknown
+{
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Download* sender,
+      [in] ICoreWebView2DownloadStateChangedEventArgs* args);
+}
+
+/// Event args for the `DownloadStateChanged` event.
+[uuid(28E3F13A-4DB1-4738-A0A7-FC4B6F6765AD), object, pointer_default(unique)]
+interface ICoreWebView2DownloadStateChangedEventArgs : IUnknown
+{
+  /// The reason why connection with file host was broken.
+  [propget] HRESULT InterruptReason(
+      [out, retval] COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON* interruptReason);
+}
+
+/// Represents a download process. Gives access to the download's metadata
+/// and supports a user canceling, pausing, or resuming the download.
+[uuid(0C4A07B1-B610-459F-A574-EC253D5EC40D), object, pointer_default(unique)]
+interface ICoreWebView2Download : IUnknown
+{
+  /// Add an event handler for the `DownloadProgressSizeInBytesChanged` event.
+  HRESULT add_DownloadProgressSizeInBytesChanged(
+    [in] ICoreWebView2DownloadProgressSizeInBytesChangedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadProgressSizeInBytesChanged`.
+  HRESULT remove_DownloadProgressSizeInBytesChanged(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `DownloadEstimatedEndTimeChanged` event.
+  HRESULT add_DownloadEstimatedEndTimeChanged(
+    [in] ICoreWebView2DownloadEstimatedEndTimeChangedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadEstimatedEndTimeChanged`.
+  HRESULT remove_DownloadEstimatedEndTimeChanged(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `DownloadStateChanged` event.
+  HRESULT add_DownloadStateChanged(
+    [in] ICoreWebView2DownloadStateChangedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadStateChanged`.
+  HRESULT remove_DownloadStateChanged(
+      [in] EventRegistrationToken token);
+
+  /// The URI of the download.
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The Content-Disposition header value from HTTP response.
+  [propget] HRESULT ContentDisposition([out, retval] LPWSTR* contentDisposition);
+
+  /// MIME type of the downloaded content.
+  [propget] HRESULT MimeType([out, retval] LPWSTR* mimeType);
+
+  /// The expected size of the download in total number of bytes based on the
+  /// HTTP content-length header. Returns -1 if the size is unknown.
```

Comment:
-1 but the type is unsigned.

Reply:
I think the type should be unsigned though right? What is the maximum size of download supported? Or what is the type that chromium is using to tell us the size? Is -1 what chromium is giving us in this case?

Reply:
Unresolving conversation since its still signed int.

Reply:
I made a mistake with unsigned int, the type in chromium is signed int for both received bytes and total bytes, and chromium gives us the -1 for total bytes if size is unknown.